packages/@react-spectrum/dialog/docs/DialogTrigger.mdx

```diff
@@ -0,0 +1,461 @@
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License. -->
+
+import {Layout} from '@react-spectrum/docs';
+export default Layout;
+
+import docs from 'docs:@react-spectrum/dialog';
+import {ExampleImage, HeaderInfo, PropTable} from '@react-spectrum/docs';
+import packageData from '../package.json';
+
+```jsx import
+import {ActionButton, Button} from '@react-spectrum/button';
+import {ButtonGroup} from '@react-spectrum/buttongroup';
+import {Checkbox} from '@react-spectrum/checkbox';
+import {Content, Footer, Header} from '@react-spectrum/view';
+import {Dialog, DialogTrigger} from '@react-spectrum/dialog';
+import {Divider} from '@react-spectrum/divider';
+import {Form} from '@react-spectrum/form';
+import {Heading, Text} from '@react-spectrum/typography';
+import {TextField} from '@react-spectrum/textfield';
+```
+
+# DialogTrigger
+
+<p>{docs.exports.DialogTrigger.description}</p>
+
+<HeaderInfo
+  packageData={packageData}
+  componentNames={['DialogTrigger']} />
+
+## Example
+
+```tsx example
+<DialogTrigger type="popover">
+  <ActionButton>Disk Status</ActionButton>
+  <Dialog>
+    <Heading>C://</Heading>
+    <Divider />
+    <Content>
+      <Text>
+        50% disk space remaining.
+      </Text>
+    </Content>
+  </Dialog>
+</DialogTrigger>
+```
+
+## Content
+
+The DialogTrigger accepts exactly two children: the element which triggers the opening
+of the Dialog and the Dialog itself. The trigger must be the first child 
+passed into the DialogTrigger and should be an element that supports press events.
+
+If your Dialog has buttons within it that should close the Dialog when pressed,
+you must wrap the Dialog in a function in order to properly propagate the DialogTrigger's
+`close` state setter to the Dialog's children. Dialogs that do not contain such interactive
+elements can simply provide the Dialog component as is to the DialogTrigger as its second child.  
+
+The example below demonstrates how to pass the DialogTrigger's `close` state setter to the Dialog's buttons. 
+
+```tsx example
+<DialogTrigger>
+  <ActionButton>Checkout</ActionButton>
+  {(close) => ( 
+    <Dialog>
+      <Heading>Confirm checkout?</Heading>
+      <Divider />
+      <Content>
+        <Text>
+          You have 5 items in your cart. Proceed to checkout?
+        </Text>
+      </Content>
+      <ButtonGroup>
+        <Button variant="secondary" onPress={close}>Cancel</Button>
+        <Button variant="cta" onPress={close} autoFocus>Confirm</Button>
+      </ButtonGroup>
+    </Dialog>
+  )}
+</DialogTrigger>
+```
+
+## Dialog Types
+
+By providing a `type` prop, you can specify the type of Dialog that is rendered
+by your DialogTrigger. 
+
+### Modal
+
+Modal Dialogs create an underlay that blocks access to the underlying user interface
+until the Dialog is closed. Sizing options can be found in the [Dialog page](./Dialog.html#size).
```

Comment:
should we talk about the focus trap as well?

Reply:
Sure I'll add a sentence about that